src/main/java/com/smartcar/sdk/Vehicle.java

```diff
@@ -126,6 +126,54 @@ public String vin() throws SmartcarException {
     return this.call("permissions", "GET", null, ApplicationPermissions.class).getData().getPermissions();
   }
 
+  /**
+   * Send request to the /permissions endpoint
+   *
+   * @param permission the permission to check on the vehicle
+   *
+   * @return if vehicle has specified permission
+   * @throws SmartcarException if the request is unsuccessful
+   */
+  public boolean hasPermissions(String permission) throws SmartcarException {
+    try {
+      for(String vehiclePermission: this.permissions()){
```

Comment:
can you pull out the `this.permissions()` out of the for loop to make it easier to read?.

Also, it would be easier to read/understand if we have a filtering function or isSubset function.
https://docs.oracle.com/javase/1.5.0/docs/api/java/util/Collection.html#containsAll(java.util.Collection)
https://www.mkyong.com/java8/java-8-streams-filter-examples/

Reply:
We can't use filters since we are backwards compatible with Java 7, using containsall